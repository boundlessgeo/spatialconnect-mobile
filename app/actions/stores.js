'use strict';
import { stores } from 'spatialconnect/native';

export const loadStores = () => {
  return dispatch => {
    stores().take(1).subscribe(data => {
      dispatch({
        type: 'ADD_STORE_LIST',
        stores: data.stores
      });
    });
  };
};