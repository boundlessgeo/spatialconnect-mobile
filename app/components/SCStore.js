'use strict';
import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View
} from 'react-native';
import { Actions } from 'react-native-router-flux';
import palette from '../style/palette';

class SCStore extends Component {

  goToMap() {
    Actions.storeMap({storeInfo: this.props.storeInfo});
  }

  render() {
    return (
      <View style={styles.container}>
        <Text numberOfLines={1}>
          Name: {this.props.storeInfo.name}
        </Text>
        <Text numberOfLines={1}>
          Type: {this.props.storeInfo.type}
        </Text>
        <Text numberOfLines={1}>
          Service: {this.props.storeInfo.service}
        </Text>
        <Text numberOfLines={0}>
          ID: {this.props.storeInfo.storeId}
        </Text>
        <Text style={styles.link} onPress={this.goToMap.bind(this)}>View on Map</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 0,
    padding: 10,
    backgroundColor: palette.gray
  },
  link: {
    color: 'blue',
    textDecorationLine: 'underline'
  }
});

export default SCStore;