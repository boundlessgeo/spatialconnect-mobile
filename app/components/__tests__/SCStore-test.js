/*global jest,describe,it,expect*/
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SCStore from '../SCStore';
import stores from 'stores';

describe('<SCStore />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <SCStore storeInfo={stores[0]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});