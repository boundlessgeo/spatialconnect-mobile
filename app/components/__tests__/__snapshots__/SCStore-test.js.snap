exports[`<SCStore /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e4e7ec",
      "flex": 1,
      "marginTop": 0,
      "padding": 10
    }
  }>
  <Text
    numberOfLines={1}>
    Name: 
    gj1
  </Text>
  <Text
    numberOfLines={1}>
    Type: 
  </Text>
  <Text
    numberOfLines={1}>
    Service: 
  </Text>
  <Text
    numberOfLines={0}>
    ID: 
  </Text>
</View>
`;
